### DISCUSSION TITLE
CommandLine Immediately Return Result

### DISCUSSION THREAD

**Discussion by ghost** [remkop/picocli#1810] [Q&A] [answered]
i have command line application that have ability to listen to port for incoming updates there is subcommand as method called `bionic listen -p <port>` in main method I have `System.exit(new CommandLine(new CheckSum()).execute(args))` when i run the command, server starts and immediately exit, how can I stop this behavior? (new CommandLine(new CheckSum() returns immediately after commands run)

**Comment by remkop** [comment, marked as answer]
Sorry for the late reply, I missed this.

One idea is to simply not return from the `run` method after starting the server.

```java
 public void run() {
    Server server = new Server();
    server.start();
    while (server.isRunning()) {
        Thread.sleep(1);
    }
}
```

**Comment by ghost** [comment]
thank you for answering, I really appreciate it
currently, I did not return anything from the run method and it works perfectly but the downside is that other commands also can't set error code, and I'm looking for a way to work around this 💭

**Reply by remkop** [reply]
Sorry, I don’t understand what the problem is. Did you see this section in the user manual? https://picocli.info/#_generating_an_exit_code

**Reply by ghost** [reply]
I must read the documentation more precisely 😀, Thank you, that was exactly what I looking for.
